apps/framework-cli/src/framework/python/parser.rs

```diff
@@ -0,0 +1,465 @@
+                        _ => {
+                            return Err(PythonParserError::UnsupportedDataTypeError {
+                                type_name: "Unsupported data type".to_string(),
+                            })
+                        }
+                    });
+                    column.data_type = Some(col_type);
+                }
+                "Key" => match &*subscript.slice {
+                    Expr::Name(name) => {
+                        let col_type = name_node_to_base_column_type(name.clone())?;
+                        column.data_type = Some(col_type);
+                        column.required = Some(true);
+                        column.primary_key = Some(true);
+                    }
+                    _ => {
+                        return Err(PythonParserError::UnsupportedDataTypeError {
+                            type_name: "Unsupported data type".to_string(),
+                        })
+                    }
+                },
+                "Optional" => match &*subscript.slice {
+                    Expr::Name(name) => {
+                        let col_type = name_node_to_base_column_type(name.clone())?;
+                        column.data_type = Some(col_type);
+                        column.required = Some(false);
+                    }
+                    _ => {
+                        return Err(PythonParserError::UnsupportedDataTypeError {
+                            type_name: "Unsupported data type".to_string(),
+                        })
+                    }
+                },
+
+                _ => {
+                    return Err(PythonParserError::UnsupportedDataTypeError {
+                        type_name: "Unsupported data type".to_string(),
+                    })
+                }
+            },
+            _ => {
+                return Err(PythonParserError::UnsupportedDataTypeError {
+                    type_name: "Unsupported data type".to_string(),
+                })
+            }
+        },
+
+        _ => {
+            return Err(PythonParserError::UnsupportedDataTypeError {
+                type_name: "Unsupported data type".to_string(),
+            })
+        }
+    }
+
+    Ok(column)
+}
+
+fn slice_node_to_column_type(slice_node: &ast::Expr) -> Result<ColumnType, PythonParserError> {
+    match slice_node {
+        Expr::Name(name) => match name.id.to_string().as_str() {
+            "str" => Ok(ColumnType::String),
+            "int" => Ok(ColumnType::Int),
+            "float" => Ok(ColumnType::Float),
+            "bool" => Ok(ColumnType::Boolean),
+            "datetime" => Ok(ColumnType::DateTime),
+            _ => Err(PythonParserError::UnsupportedDataTypeError {
+                type_name: name.id.to_string(),
+            }),
+        },
+        _ => Err(PythonParserError::UnsupportedDataTypeError {
+            type_name: "Unsupported data type".to_string(),
```

Comment:
Here you should put the name of the type that's missing. The sentence `Unsupported data Type` will be provided by the error string that you templated above.

Reply:
Maybe here - the error should be a different one?